_drafts/SC3-3-2-form-field-has-name.md

```diff
@@ -0,0 +1,78 @@
+---
+name: Form field has accessible name
+description: |
+  Each form field element has an accessible name
+
+success_criterion:
+- 3.3.2 # Labels or Instructions
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Anne Thyme Nørregaard
+- Bryn Anderson
+---
+
+## Test procedure
+
+### Applicability
+
+The rules applies to:
+- any element that is or [exposed to assistive technologies](#exposed-to-assistive-technologies),
+- that has a [semantic role](#semantic-role) that inherits from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and 
```

Comment:
This has two meanings, and so it's not objective (as required by ACT-RF). This can mean either roles that inherit from select or input in ARIA 1.1, or roles defined in any future standard (regardless of who published it!) and use ARIA 1.1 select to inherit from.

It seems to me this should only be the first one... although a better solution IMO is to just use the list from the note. That solves "plain language" as well. I realise this is trying to be future proof, but in my view it's doing quite the opposite, tying this to the inheritance in ARIA 1.1. If in ARIA 1.2 `menuitemcheckbox` or some other role is changed to inherit from something else, this rule has to be updated to fit.

Reply:
When reading the rule I was only thinking about ARIA 1.1. But Wilco's comment makes all the sense.

Reply:
I do not agree that hard coding a list of roles that only the rule authors know the reasoning behind is a good idea. That's a maintenance nightmare waiting to happen. As for limiting it to ARIA 1.1, I don't really see the point. If ARIA 1.2 indeed changes a role that was previously considered a form control to inherit from something else, then the rule should no longer apply to that role. All form controls inherit from either `input` or `select` and I don't see that changing in the future. In the contrary, if a role is changed in ARIA 1.2 to inherit from `input` or `select` then the rule needs no further changes to accommodate this.

Reply:
@WilcoFiers @carlosapaduarte can you please comment on kasper's reply.